presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/BridgingHiveMetastore.java

```diff
@@ -116,9 +116,9 @@ public PartitionStatistics getTableStatistics(HiveIdentity identity, Table table
     }
 
     @Override
-    public void updateTableStatistics(HiveIdentity identity, String databaseName, String tableName, Function<PartitionStatistics, PartitionStatistics> update)
+    public void updateTableStatistics(HiveIdentity identity, Table table, Function<PartitionStatistics, PartitionStatistics> update)
     {
-        delegate.updateTableStatistics(identity, databaseName, tableName, update);
+        delegate.updateTableStatistics(identity, toMetastoreApiTable(table), update);
```

Comment:
As discussed offline. I am not sure where it's ok to repopulate metastore based on `toMetastoreApiTable` results.

@electrum are we losing information?

Reply:
@wendigo , i talked with @electrum about this. Conclusion was -- this is not safe to persist a roundrip of `fromMetastoreApiTable + toMetastoreApiTable`. 
Instead, we should find a workaround for Avro tables that would get us going. Or do additional call to metastore for Avro tables.

Let's discuss further.